be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -546,12 +546,7 @@ Status FragmentExecutor::_decompose_data_sink_to_operator(RuntimeState* runtime_
         if (sender->get_partition_type() == TPartitionType::HASH_PARTITIONED ||
             sender->get_partition_type() == TPartitionType::BUCKET_SHUFFLE_HASH_PARTITIONED) {
             dest_dop = t_stream_sink.dest_dop;
-
-            // UNPARTITIONED mode will be performed if both num of destination and dest dop is 1
-            // So we only enable pipeline level shuffle when num of destination or dest dop is greater than 1
-            if (sender->destinations().size() > 1 || dest_dop > 1) {
```

Comment:
Why change this logic?

Reply:
This condition here is redundant, ExchangeSinkOperator's ctor will take care of both of them, including destination num and dest dop

```cpp
ExchangeSinkOperator::ExchangeSinkOperator(OperatorFactory* factory, int32_t id, int32_t plan_node_id,
                                           int32_t driver_sequence, const std::shared_ptr<SinkBuffer>& buffer,
                                           TPartitionType::type part_type,
                                           const std::vector<TPlanFragmentDestination>& destinations,
                                           bool is_pipeline_level_shuffle, const int32_t num_shuffles_per_channel,
                                           int32_t sender_id, PlanNodeId dest_node_id,
                                           const std::vector<ExprContext*>& partition_expr_ctxs,
                                           bool enable_exchange_pass_through, FragmentContext* const fragment_ctx,
                                           const std::vector<int32_t>& output_columns)
        : Operator(factory, id, "exchange_sink", plan_node_id, driver_sequence),
          _buffer(buffer),
          _part_type(part_type),
          _destinations(destinations),
          _num_shuffles_per_channel(num_shuffles_per_channel > 0 ? num_shuffles_per_channel : 1),
          _sender_id(sender_id),
          _dest_node_id(dest_node_id),
          _partition_expr_ctxs(partition_expr_ctxs),
          _fragment_ctx(fragment_ctx),
          _output_columns(output_columns) {
...
    _num_shuffles = _channels.size() * _num_shuffles_per_channel;

    _is_pipeline_level_shuffle = is_pipeline_level_shuffle && (_num_shuffles > 1);
}
```